File: java/src/main/java/vsdk/toolkit/io/context/ColorRegistryContext.java
package vsdk.toolkit.io.context;

import vsdk.toolkit.common.dataStructures.LookUpBehaviors;
import vsdk.toolkit.common.dataStructures.LookUpTable;

public class ColorRegistryContext {
    public LookUpTable<ColorContext> colorLookUpTable;
    public ColorContext unNamedColorContext;
    public ColorContext currentColor;

    public ColorRegistryContext() {
        colorLookUpTable = new LookUpTable<>(LookUpBehaviors.OWNING);
        unNamedColorContext = new ColorContext();
        currentColor = unNamedColorContext;
        unNamedColorContext.copy(ColorContext.DEFAULT_COLOR_CONTEXT);
    }

    public void destroy() {
        unNamedColorContext = null;
        if (colorLookUpTable != null) {
            colorLookUpTable.lookUpDone();
            colorLookUpTable = null;
        }
        currentColor = null;
    }

    public void reset() {
        unNamedColorContext.copy(ColorContext.DEFAULT_COLOR_CONTEXT);
        currentColor = unNamedColorContext;
        colorLookUpTable.lookUpDone();
    }
}

File: typescript/src/main/vsdk/toolkit/io/context/ColorRegistryContext.ts
import { LookUpBehaviors } from "../../common/dataStructures/LookUpBehaviors";
import { LookUpTable } from "../../common/dataStructures/LookUpTable";
import { ColorContext } from "./ColorContext";

export class ColorRegistryContext {
  public colorLookUpTable: LookUpTable<ColorContext> | null;
  public unNamedColorContext: ColorContext | null;
  public currentColor: ColorContext | null;

  public constructor() {
    this.colorLookUpTable = new LookUpTable<ColorContext>(LookUpBehaviors.OWNING);
    this.unNamedColorContext = new ColorContext();
    this.currentColor = this.unNamedColorContext;
    this.unNamedColorContext.copy(ColorContext.DEFAULT_COLOR_CONTEXT);
  }

  public destroy(): void {
    this.unNamedColorContext = null;
    if (this.colorLookUpTable !== null) {
      this.colorLookUpTable.lookUpDone();
      this.colorLookUpTable = null;
    }
    this.currentColor = null;
  }

  public reset(): void {
    (this.unNamedColorContext as ColorContext).copy(ColorContext.DEFAULT_COLOR_CONTEXT);
    this.currentColor = this.unNamedColorContext;
    (this.colorLookUpTable as LookUpTable<ColorContext>).lookUpDone();
  }
}
